### ISSUE TITLE
Required package: svelte (via "svelte\compiler")

### ISSUE THREAD

**Issue by rtritto** [sveltejs/eslint-plugin-svelte#1561]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

10.7.0

### What version of `eslint-plugin-svelte` are you using?

3.21.0

### What did you do?

<details>
<summary>Configuration</summary>
import { defineConfig } from 'eslint/config'
import pluginTypescript from 'typescript-eslint'
import svelte from 'eslint-plugin-svelte'

export default defineConfig(
  {
    extends: [
      pluginTypescript.configs.recommended
    ],
    rules: {
      'comma-dangle': [1, 'never'],
      'semi': [1, 'never'],

      '@typescript-eslint/no-explicit-any': 0,
      '@typescript-eslint/no-unused-vars': [1, { argsIgnorePattern: '^_' }],
    }
  },
  {
    plugins: { svelte },
    files: ['packages/vike-lite-svelte/**/*.{ts,svelte}']
  }
)
</details>

### What did you expect to happen?

Lint completed

### What actually happened?

```sh
Oops! Something went wrong! :(

ESLint: 10.7.0

Error: eslint-plugin-svelte tried to access svelte (a peer dependency) but it isn't provided by your application; this makes the require call ambiguous and unsound.

Required package: svelte (via "svelte\compiler")
Required by: eslint-plugin-svelte@virtual:9402575293a0974a3839801def265e465fa67b894c3f14fcae1c96d15b3f3dfbc81dcb06893b26af368e4ff19f4c44c5d694420d65ad987a9ad3f3bfdc1ace7b#npm:3.21.0 (via C:\node-ecosystem\vike-lite\.yarn\__virtual__\eslint-plugin-svelte-virtual-1beae2387d\5\Users\<USER>\AppData\Local\Yarn\Berry\cache\eslint-plugin-svelte-npm-3.21.0-ff2a35f01e-10c0.zip\node_modules\eslint-plugin-svelte\lib\utils\svelte-context.js)
Ancestor breaking the chain: vike-lite-monorepo@workspace:.


    at makeError (C:\node-ecosystem\vike-lite\.pnp.cjs:14771:34)
    at resolveToUnqualified (C:\node-ecosystem\vike-lite\.pnp.cjs:16356:21)
    at resolveRequest (C:\node-ecosystem\vike-lite\.pnp.cjs:16545:31)
    at Object.resolveRequest (C:\node-ecosystem\vike-lite\.pnp.cjs:16608:26)
    at resolve$1 (file:///C:/node-ecosystem/vike-lite/.pnp.loader.mjs:2046:21)
    at nextResolve (node:internal/modules/esm/hooks:785:28)
    at AsyncLoaderHooksOnLoaderHookWorker.resolve (node:internal/modules/esm/hooks:269:30)
    at MessagePort.handleMessage (node:internal/modules/esm/worker:251:24)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:845:20)
    at MessagePort.<anonymous> (node:internal/per_context/messageport:23:28)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

- clone https://github.com/node-ecosystem/vike-lite/tree/bug-eslint-svelte
- `yarn`
- `yarn lint:svelte`

### Additional comments

Note: `svelte` needs to remain as optional peer dependency (peer dependency meta true). Perhaps `svelte\compiler` should be used only if `svelte` is installed.

**Comment by rtritto** [comment]
FYI @baseballyama

**Comment by baseballyama** [comment]
The README says to install Svelte as well.
https://github.com/sveltejs/eslint-plugin-svelte#manual-setup

And Svelte is already in `peerDependencies`.
https://github.com/sveltejs/eslint-plugin-svelte/blob/0d9f1a057ba545313484c1f3414cf70205cb6db6/packages/eslint-plugin-svelte/package.json#L53-L56

**Comment by rtritto** [comment]
@baseballyama if `svelte` is a required peerDependency, you have to remove it from peerDependenciesMeta.
